config/gulp/file-prep.js

```diff
@@ -0,0 +1,255 @@
+  var filetype = "";
+  var paths = filePaths;
+
+  fs.readdir(paths.uploads, (err, files) => {
+    // process.stdout.write(files.length.toString() + "\n");
+    for (var file of files) {
+      if (allExtensions.includes(path.extname(file))) {
+        // creates new normalized file name
+        newfileName = cleanFileName(file);
+        filetype = fileType(file);
+        // create working directories if they do not exist
+        createDir(paths[filetype].toProcess, 3);
+        if (filetype === "image") createDir(paths[filetype].processed, 3);
+        // copies uploaded file to /to-process with new normalized name
+        fs.renameSync(
+          paths.uploads + "/" + file,
+          paths[filetype].toProcess + "/" + newfileName
+        );
+      }
+    }
+    if (err) {
+      process.output.write(
+        "Error cleaning and copying file [" + file + "]\n",
+        "Error message: " + err.message
+      );
+    }
+  });
+  done();
+}
+
+
+
+/**
+ * creates the originals and to-process directories for both files and images
+ * ./content/uploads/_working-images/to-process";
+ * ./content/uploads/_working-images/processed";
+ * @param {string} directoryPath - path of directory to create
+ * @param {number} foldersDeep - depth of directory relative to base project 
+ */
+function createDir(directoryPath, foldersDeep) {
+  var dp = filePaths.base;
+
+  //if this directory does not exist, create it
+  if (!fs.existsSync(directoryPath)) {
+    //split directory folders
+    var subdir = directoryPath.split("/").slice(foldersDeep);
+    //create parent directories first
+    for (let i = 0; i < subdir.length; i++) {
+      for (let j = i; j <= i; j++) {
+        dp = dp + subdir[j] + "/";
+        console.log("i: " + i + " , j: " + j + "\n" + dp);
+        //check if subfolder exists
+        if (fs.existsSync(dp)) {
+          continue;
+        }
+        // create folder
+        else {
+          fs.mkdirSync(dp, (err) => {
+            "Error creating subdirectory [" + subdir[i] + "]\n",
+              "Error message: " + err.message;
+          });
+        }
+      }
+    }
+  }
+}
+
+/**
+ * Checks the file extension and returns a string value of file or image
+ * @param {String} extension - file name extension (.pdf, .png, etc...)
+ * @returns a string value of image or file
+ */
+function fileType(extension) {
+  if (fileRegex.test(extension)) return "file";
+  if (imageRegex.test(extension)) return "image";
+}
+
+/**
+ * Normalizes filename by removing unnecessary characters
+ * @param {string} origfilename - filename of original uploaded file
+ * @example 
+ * File Name.pdf returns file-name.pdf
+ * @returns filename in string format
+ */
+function cleanFileName(origfilename) {
+  return origfilename
+    .toLowerCase()
+    .replace(/[ &$_#!?.]/g, "-")
+    .replace(/-+/g, "-") // multiple dashes to a single dash
+    .replace(/-(png|jpg|jpeg|pdf|doc|docx|ppt|pptx|pptm|xls|xlsx)/g, ".$1") // remove trailing dashes
```

Comment:
I've created a regex on line 26 from the list of extensions, but when using it on line 140, the filename does not get updated and reads as `test-upload-7-doc` instead of `test-upload-7.doc`. Any suggestions?